web/modules/custom/search_api_field/src/Plugin/Field/FieldFormatter/SearchFormatter.php

```diff
@@ -0,0 +1,294 @@
+      }
+      // Search results might be stale, so we check if the entity has been
+      // found in the system.
+      if (!$entity) {
+        continue;
+      }
+
+      // Avoid recursions when an entity contains another search field.
+      $entity->do_not_recurse = TRUE;
+
+      // Use the view mode configured in the field type settings or fallback
+      // to default view mode.
+      $entity_type = $entity->getEntityTypeId();
+      $entity_bundle = $entity->bundle();
+      $datasource_id = 'entity:' . $entity_type;
+      if (!empty($view_mode_settings[$datasource_id][$entity_bundle])) {
+        $view_mode = $view_mode_settings[$datasource_id][$entity_bundle];
+      }
+      else {
+        $view_mode = 'default';
+      }
+
+      $results[] = [
+        '#theme' => 'search_api_field_result',
+        '#item' => $this->entityTypeManager->getViewBuilder($entity_type)->view($entity, $view_mode),
+        '#entity' => $entity,
+      ];
+    }
+
+    $build = [
+      '#theme' => 'search_api_field',
+    ];
+
+    if (!empty($results)) {
+      $build += [
+        '#search_title' => [
+          '#markup' => $this->t('Search results'),
+        ],
+        '#no_of_results' => [
+          '#markup' => $this->formatPlural($result->getResultCount(), '1 result found', '@count results found'),
+        ],
+        '#results' => $results,
+        '#pager' => [
+          '#type' => 'pager',
+        ],
+      ];
+
+      // Build pager.
+      pager_default_initialize($result->getResultCount(), $limit);
+    }
+    else {
+      $build['#no_results_found'] = array(
+        '#markup' => $this->t('Your search yielded no results.'),
+      );
+
+      $build['#search_help'] = array(
+        '#markup' => $this->t('<ul>
+<li>Check if your spelling is correct.</li>
+<li>Remove quotes around phrases to search for each word individually. <em>bike shed</em> will often show more results than <em>&quot;bike shed&quot;</em>.</li>
+<li>Consider loosening your query with <em>OR</em>. <em>bike OR shed</em> will often show more results than <em>bike shed</em>.</li>
+</ul>'),
+      );
+    }
+
+    return $build;
+  }
+
+  /**
+   * Applies query presets configured in the field instance.
+   *
+   * @param \Drupal\search_api\Query\QueryInterface $query
+   *   The search query object.
+   * @param string $presets
+   *   The presets string.
+   */
+  protected function applyPresets(QueryInterface $query, $presets) {
+    $list = explode("\n", $presets);
+    $list = array_map('trim', $list);
+    $list = array_filter($list, 'strlen');
+
+    foreach ($list as $line) {
+      $matches = [];
+      if (preg_match('/([^\|]*)\|([^\|]*)(?:\|(.*))?/', $line, $matches)) {
+        $field = trim($matches[1]);
+        $value = trim($matches[2]);
+        $operator = !empty($matches[3]) ? trim($matches[3]) : '=';
+
+        if ($operator === 'IN') {
+          $value = explode(',', $value);
+          $value = array_map('trim', $value);
+        }
```

Comment:
Can you maybe provide a bit of inline documentation in this method to make it a bit clearer what the intention is? Thanks!